sqlglot/schema.py

```diff
@@ -96,6 +96,27 @@ def supported_table_args(self) -> t.Tuple[str, ...]:
 
         return self._supported_table_args
 
+    def normalize(self) -> None:
```

Comment:
make this private and call it in the iniatilizer have it take an argument of mapping and return a mapping

Reply:
That's how I initially approached it, but I thought that maybe it's too opinionated to implicitly change the user-supplied schema @ init time? Also, there were some problems with the executor, because `Tables` inherits from `AbstractMappingSchema` and so it would implicitly use that normalization logic.

Are we sure we want to have this be the default behavior in `__init__`?

Reply:
i don't understand why you wouldn't want it

Reply:
Because there might be another use case (besides optimization) where the user doesn't want their schema changed under the hood, for whatever reason. As I see it, the problem right now is local to the optimizer, but that's only because we change the identifiers to lowercase anyway. No problem with moving it back to `__init__`, though.

Reply:
> Tables inherits from AbstractMappingSchema and so it would implicitly use that normalization logic.

I think we need similar logic for `Tables`, right? We just might not want to transform the column names, since that might be slow...

We can also just move this to `MappingSchema` for now, which would avoid any conflict with `Tables`.

Reply:
Got it, I'll try to temporarily move it to `MappingSchema` and take another pass when the other issues are resolved and see if it can be improved.